File: CRUD/VB.Net/Aplicacion con SQL Server V2/Sistema/Sistema/Functions/Funciones.vb
' Written By Ismael Heredia in the year 2017

Imports System
Imports System.Collections.Generic
Imports System.Text

Class Funciones
    Public Function md5_encode(input As String) As String

        Using md5 As System.Security.Cryptography.MD5 = System.Security.Cryptography.MD5.Create()
            Dim inputBytes() As Byte = System.Text.Encoding.ASCII.GetBytes(input)
            Dim hashBytes() As Byte = md5.ComputeHash(inputBytes)
            Dim sb As New StringBuilder()
            For i As Integer = 0 To hashBytes.Length - 1
                sb.Append(hashBytes(i).ToString("X2"))

            Next
            Return sb.ToString().ToLower()

        End Using

    End Function

    Public Function valid_number(numberString As String) As Boolean
        Dim number As Integer
        Return Integer.TryParse(numberString, number)
    End Function

    Public Function fecha_del_dia() As String
        Dim Hoy As DateTime = DateTime.Today
        Dim fecha_actual As String = Hoy.ToString("yyyy-MM-dd")
        Return fecha_actual

    End Function


End Class

File: CRUD/C#/Sistema en ASP.Net con SQL Server/Sistema/Sistema/Functions/Funciones.cs
﻿// Written By Ismael Heredia in the year 2017

using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Text;

namespace Sistema
{
    public class Funciones
    {

        public string mensaje(string texto)
        {
            return "<script>alert('"+texto+"');</script>";
        }

        public string mensaje_con_redireccion(string texto, string ruta)
        {
            return "<script>" +
                    "alert('"+texto+"');" +
                    "window.location.href = '" + ruta + "';" +
                    "</script>";
        }

        public string get_username_in_cookie(string cookie_content)
        {
            string[] datos = cookie_content.Split('-');
            string username_cookie = datos[0];
            string password_cookie = datos[1];
            return username_cookie;
        }

        public bool valid_cookie(string content)
        {
            UsuarioDatos usuarioDatos = new UsuarioDatos();
            string[] data = content.Split('-');
            string username = data[0];
            string password = data[1];
            if (username != "" && password != "")
            {
                if (usuarioDatos.check_login(username, password))
                {
                    return true;
                }
                else
                {
                    return false;
                }
            }
            else
            {
                return false;
            }
        }

        public bool valid_cookie_admin(string content)
        {
            UsuarioDatos usuarioDatos = new UsuarioDatos();
            string[] data = content.Split('-');
            string username = data[0];
            string password = data[1];
            if (username != "" && password != "")
            {
                if (usuarioDatos.check_login(username, password))
                {
                    if (usuarioDatos.get_user_type(username) == "Administrador")
                    {
                        return true;
                    }
                    else
                    {
                        return false;
                    }
                }
                else
                {
                    return false;
                }
            }
            else
            {
                return false;
            }
        }

        public string md5_encode(string input)
        {
            using (System.Security.Cryptography.MD5 md5 = System.Security.Cryptography.MD5.Create())
            {
                byte[] inputBytes = System.Text.Encoding.ASCII.GetBytes(input);
                byte[] hashBytes = md5.ComputeHash(inputBytes);

                StringBuilder sb = new StringBuilder();
                for (int i = 0; i < hashBytes.Length; i++)
                {
                    sb.Append(hashBytes[i].ToString("X2"));
                }
                return sb.ToString().ToLower();
            }
        }

        public bool valid_number(String numberString)
        {
            int number;
            return int.TryParse(numberString, out number);
        }

        public String fecha_del_dia()
        {
            DateTime Hoy = DateTime.Today;
            string fecha_actual = Hoy.ToString("yyyy-MM-dd");
            return fecha_actual;
        }
    }
}